tap_github/authenticator.py

```diff
@@ -156,52 +227,40 @@ def prepare_tokens(self) -> List[TokenManager]:
             # Accept multiple tokens using environment variables GITHUB_TOKEN*
             env_tokens = {
                 value
-                for key, value in environ.items()
+                for key, value in env_dict.items()
                 if key.startswith("GITHUB_TOKEN")
             }
             if len(env_tokens) > 0:
                 self.logger.info(
                     f"Found {len(env_tokens)} 'GITHUB_TOKEN' environment variables for authentication."
                 )
-                personal_tokens = env_tokens
+                personal_tokens = personal_tokens.union(env_tokens)
```

Comment:
⚠️ Note: this is a functionality change!  I noticed this edge case:
- `auth_token` is provided in the config
- `additional_auth_tokens` is not provided in the config
- environment variables whose keys start with `GITHUB_TOKEN` are present

The previous logic would have meant that the detected environment variables were used, but `auth_token` wasn't.  I can't imagine we want to ignore a token specified in the config, so I am assuming this is a bug.  To be honest, my preference would be that if anything is specified in the config then all personal tokens in the environment are ignored, but it's possible that could be a potentially breaking change for some users.

(Let me know if you want me to move this change to a different PR, or change this PR's type to `bug` instead of `refactor`. :P)

Reply:
That's indeed a bug 😬. More then happy to communicate this refactor as a bugfix. Thanks for chasing it 🙏.

Reply:
Done!